core/unit_test/TestAtomicAccessor.hpp

```diff
@@ -0,0 +1,67 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <Kokkos_Core.hpp>
+#include <type_traits>
+
+#include <gtest/gtest.h>
+
+template <class T, class ExecutionSpace>
+void test_atomic_accessor() {
+  using value_type = std::remove_const_t<T>;
+  Kokkos::View<value_type*, ExecutionSpace> v("V", 100);
+
+  Kokkos::parallel_for(
+      Kokkos::RangePolicy<ExecutionSpace>(0, v.extent(0)),
+      KOKKOS_LAMBDA(int i) { v(i) = i; });
+
+  int errors;
+  using acc_t = Kokkos::Impl::AtomicAccessorRelaxed<T>;
+  acc_t acc{};
+  typename acc_t::data_handle_type ptr = v.data();
+
+  Kokkos::parallel_reduce(
+      Kokkos::RangePolicy<ExecutionSpace>(0, v.extent(0)),
+      KOKKOS_LAMBDA(int i, int& error) {
+        if (acc.access(ptr, i) != ptr[i]) error++;
+        if (acc.offset(ptr, i) != ptr + i) error++;
```

Comment:
Wanna load the value and assert it is `i` or make some other transaction?

Reply:
no, I trust `ptr[i]` is the right thing.